azure-devops/projects/io-pay-projects/io-pay-portal-fe.tf

```
@@ -0,0 +1,214 @@
+
+  variable {
+    name  = "CACHE_VERSION_ID"
+    value = var.io-pay-portal-fe.pipeline.cache_version_id
+  }
+
+  variable {
+    name  = "BLOB_CONTAINER_NAME"
+    value = var.io-pay-portal-fe.pipeline.blob_container_name
+  }
+
+  variable {
+    name  = "ENDPOINT_AZURE"
+    value = var.io-pay-portal-fe.pipeline.endpoint_azure
+  }
+
+  variable {
+    name  = "IO_PAY_PORTAL_API_HOST"
+    value = var.io-pay-portal-fe.pipeline.io_pay_portal_api_host
+  }
+
+  variable {
+    name  = "IO_PAY_PORTAL_API_REQUEST_TIMEOUT"
+    value = var.io-pay-portal-fe.pipeline.io_pay_portal_api_request_timeout
+  }
+
+  variable {
+    name  = "IO_PAY_PORTAL_PAY_WL_HOST"
+    value = var.io-pay-portal-fe.pipeline.io_pay_portal_pay_wl_host
+  }
+  
+  variable {
+    name  = "IO_PAY_PORTAL_PAY_WL_POLLING_ATTEMPTS"
+    value = var.io-pay-portal-fe.pipeline.io_pay_portal_pay_wl_polling_attempts
+  }
+  
+  variable {
+    name  = "IO_PAY_PORTAL_PAY_WL_POLLING_INTERVAL"
+    value = var.io-pay-portal-fe.pipeline.io_pay_portal_pay_wl_polling_interval
+  }
+  
+  variable {
+    name  = "PROFILE_NAME_CDN_AZURE"
+    value = var.io-pay-portal-fe.pipeline.profile_name_cdn_azure
+  }
+    
+  variable {
+    name  = "RESOURCE_GROUP_AZURE"
+    value = var.io-pay-portal-fe.pipeline.resource_group_azure
+  }
+
+  variable {
+    name  = "STORAGE_ACCOUNT_NAME"
+    value = var.io-pay-portal-fe.pipeline.storage_account_name
+  }
+
+}
+
+# deploy serviceendpoint authorization
+resource "azuredevops_resource_authorization" "io-pay-portal-fe-deploy-github-ro-auth" {
+  depends_on = [azuredevops_serviceendpoint_github.io-azure-devops-github-ro, azuredevops_build_definition.io-pay-portal-fe-deploy, azuredevops_project.project]
+
+  project_id    = azuredevops_project.project.id
+  resource_id   = azuredevops_serviceendpoint_github.io-azure-devops-github-ro.id
+  definition_id = azuredevops_build_definition.io-pay-portal-fe-deploy.id
+  authorized    = true
+  type          = "endpoint"
+}
+
+resource "azuredevops_resource_authorization" "io-pay-portal-fe-deploy-github-rw-auth" {
+  depends_on = [azuredevops_serviceendpoint_github.io-azure-devops-github-rw, azuredevops_build_definition.io-pay-portal-fe-deploy, azuredevops_project.project]
+
+  project_id    = azuredevops_project.project.id
+  resource_id   = azuredevops_serviceendpoint_github.io-azure-devops-github-rw.id
+  definition_id = azuredevops_build_definition.io-pay-portal-fe-deploy.id
+  authorized    = true
+  type          = "endpoint"
+}
+
+resource "azuredevops_resource_authorization" "io-pay-portal-fe-deploy-azurerm-PROD-IO-auth" {
+  depends_on = [azuredevops_serviceendpoint_azurerm.PROD-IO, azuredevops_build_definition.io-pay-portal-fe-deploy, time_sleep.wait]
+
+  project_id    = azuredevops_project.project.id
+  resource_id   = azuredevops_serviceendpoint_azurerm.PROD-IO.id
+  definition_id = azuredevops_build_definition.io-pay-portal-fe-deploy.id
+  authorized    = true
+  type          = "endpoint"
+}
```

```suggestion
}

resource "azurerm_role_assignment" "io-pay-deploy-azurerm-PROD-IO-iopstcdniopayportal" {
  depends_on = [data.azuread_service_principal.service_principals]

  principal_id         = data.azuread_service_principal.service_principals[local.PROD-IO-UID].id
  role_definition_name = "Storage Blob Data Contributor"
  scope                = "/subscriptions/${data.azurerm_key_vault_secret.key_vault_secret["TTDIO-PROD-IO-SUBSCRIPTION-ID"].value}/resourceGroups/${var.io-pay-portal-fe.pipeline.resource_group_azure}/providers/Microsoft.Storage/storageAccounts/${var.io-pay-portal-fe.pipeline.production_storage_account_name}"
}
```

Add permission to service principal to write into production_storage_account_name